src/sentry/api/endpoints/project_group_index.py

```diff
@@ -346,74 +376,112 @@ def put(self, request, project):
 
             now = timezone.now()
 
-            for group in group_list:
-                try:
-                    with transaction.atomic():
-                        resolution, created = GroupResolution.objects.create(
-                            group=group,
-                            release=release,
-                        ), True
-                except IntegrityError:
-                    resolution, created = GroupResolution.objects.get(
-                        group=group,
-                    ), False
-
-                if created:
-                    activity = Activity.objects.create(
-                        project=group.project,
-                        group=group,
-                        type=Activity.SET_RESOLVED_IN_RELEASE,
-                        user=acting_user,
-                        ident=resolution.id,
-                        data={
-                            # no version yet
-                            'version': '',
-                        }
-                    )
-                    activity.send_notification()
+            self._resolve_in_release(
+                actor=acting_user,
+                group_list=group_list,
+                release=release,
+            )
 
             queryset.update(
                 status=GroupStatus.RESOLVED,
                 resolved_at=now,
             )
 
-            result.update({
-                'status': 'resolved',
-                'statusDetails': {
-                    'inNextRelease': True,
-                },
-            })
+            result['status'] = 'resolved'
+            if status == 'resolvedInNextRelease':
+                result['statusDetails'] = {'inNextRelease': True}
+            else:
+                result['statusDetails'] = {'inThisRelease': True}
+
+        elif status == 'resolved':
+            # This resolve path has to handle three cases, and they're done
+            # in slightly odd ways since it's handling items in bulk.
+            #   1) All Groups have releases attached to them.
+            #   2) Some Groups have releases attached.
+            #   3) No Groups have releases attached.
+
+            # Fetch the latest release tags for the set of groups
+            tags = list(GroupTagValue.objects.filter(
+                group__in=group_ids,
+                key__in=('sentry:release', 'release'),
+            ).order_by('-last_seen'))
```

Comment:
This query is definitely wrong if a group contains multiple releases, it's not picking the latest. It's just sorting all of them. Need to figure out how to query "latest release for each group".

:(